include/uart.h

```diff
@@ -29,41 +29,216 @@ extern "C" {
 
 #include <device.h>
 
-#ifdef CONFIG_PCI
-#include <drivers/pci/pci.h>
-#include <drivers/pci/pci_mgr.h>
-#endif
+/** @brief Line control signals. */
+enum uart_line_ctrl {
+	UART_LINE_CTRL_RTS = (1 << 1),
+	UART_LINE_CTRL_DTR = (1 << 2),
+	UART_LINE_CTRL_DCD = (1 << 3),
+	UART_LINE_CTRL_DSR = (1 << 4),
+};
+
 /**
- * @brief Options for @a UART initialization.
+ * @brief Types of events passed to callback in UART_HIGH_LEVEL_API
+ *
+ * Receiving:
+ * 1. To start receiving, @ref rx_enable has to be called with first buffer.
+ * 2. When receiving starts to current buffer, UART_RX_BUF_REQUEST will be
+ *    generated, in response to that user can either:
+ *
+ *    - Provide second buffer using @ref rx_buf_rsp, when first buffer is
+ *      filled, receiving will automatically start to second buffer.
+ *    - Ignore the event, this way when current buffer is filled UART_RX_DONE
+ *      event will be generated and receiving will be stopped.
+ *
+ * 3. If some data was received and timeout occurred UART_RX_RDY event will be
+ *    generated. It can happen multiples times for the same buffer. RX timeout
+ *    is counted from last byte received i.e. if no data was received, there
+ *    won't be any timeout event.
+ * 4. After buffer is filled UART_RX_RDY is generated along with
+ *    UART_RX_BUF_RELEASED indicating that current buffer is no longer used.
```

Comment:
Thanks for clarifying this.  I think you should be clear on the order of these events, i.e. `UART_RX_RDY` will be generated, immediately followed by `UART_RX_BUF_RELEASED`.  "along with" leaves the order underspecified, and doing it the other way around could result in loss of data or invalid pointer dereferencing.

Reply:
Right, fixed as proposed.